src/openrct2/ride/RideData.cpp

```diff
@@ -338,9 +338,19 @@ constexpr const RideTypeDescriptor RideTypeDescriptors[RIDE_TYPE_COUNT] = {
     /* RIDE_TYPE_CLASSIC_WOODEN_ROLLER_COASTER      */ ClassicWoodenRollerCoasterRTD,
 };
 
-bool RideTypeDescriptor::HasFlag(uint64_t flag) const
+bool RideTypeDescriptor::HasFlag(RideTypeFlags flag) const
 {
-    return Flags & flag;
+    return Flags.get(static_cast<size_t>(flag));
+}
+
+bool RideTypeDescriptor::HasFlag(std::vector<RideTypeFlags> flags) const
```

Comment:
could this be va_args instead of vector?

Reply:
I prefer std::vector it has a simpler code syntax than va_args.

Reply:
std::vector however uses dynamic memory allocation which is probably not desired. As @duncanspumpkin suggested its better to enable the required operators for enum classes that act as flags then we can trivially pass this as a single value.